enhancements/proxy-cache.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Quay as a Proxy Cache
+authors:
+  - "@fmissi"
+  - "@sleesinc"
+  - "@sdadi"
+reviewers:
+  - "@fmissi"
+  - "@sleesinc"
+  - "@sdadi"
+approvers:
+  - "@fmissi"
+  - "@sleesinc"
+  - "@sdadi"
+creation-date: 2021-12-10
+last-updated: 2021-12-10
+status: implementable
+---
+
+# Quay as a cache proxy for upstream registries
+
+Container development has become widely popular. Customers today rely on container images from upstream registries(like docker, gcp) to get desired services up and running.
+Registries now have rate limitation and throttling on the number of times users can pull from these registries.
+This proposal is to enable Quay to act as a pull through cache where, once pulled images are only pulled again when upstream images have been updated.   
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+ > 1. Will the check against the upstream registry count against an upstream rate limit?
+
+## Summary
+
+Dependencies on container images has increased tremendously with the adoption of container driven development. With the introduction of rate limits
+on popular container registries, Quay will act as a proxy cache to circumvent pull rate limitation from upstream registries. Adding a cache will also
+accelerate pull performance as images are pulled from cache rather than upstream dependencies. Cached images are updated only when the upstream
+image digest differs from cached image. 
+
+### Goals
+
+* A Quay user can define and configure(credentials, staleness period) via config file/app, a repository in Quay that acts a cache for a specific upstream registry.
+* A Quay super can user leverage storage quota of an organization to limit cache size. This means when cache size reaches its quota limit,
+  images from cache are evicted based on LRU.
+* A proxy cache organization will transparently cache and stream images to client. The images in the proxy cache organization should
+  support the default expected behaviour (like security scanner, time machine, etc) as other images on Quay.
+* Given the sensitive nature of accessing potentially untrusted upstream registry all cache pulls needs to be logged (audit log).
+* A Quay user can flush the cache to eliminate excess storage consumption.
```

Comment:
I don't think this should be handled manually by the user. It should be done automatically, based on some policy that we decide on. e.g Removing the oldest cached tags/repos.

Reply:
The automatic removal will exist along side with the manual flush. Manual flush is specifically mentioned in the user stories on the [epic](https://issues.redhat.com/browse/PROJQUAY-465):
> * As a Quay admin I want to be able to leverage the storage quota of an organization to limit the cache size so that backend storage consumption remains predictable by discarding images from the cache according to least recently used or pull frequency
> * [...]
> * As a user I want to be able to flush the cache so that I can eliminate excess storage consumption
If you think this shouldn't be the case then we need to bring it up with Daniel when he's back.